src/Symfony/Component/HttpFoundation/RequestMatcher/ChainRequestMatcher.php

```diff
@@ -0,0 +1,44 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\HttpFoundation\RequestMatcher;
+
+use Symfony\Component\HttpFoundation\Request;
+
+/**
+ * @author Roland Franssen <franssen.roland@gmail.com>
+ */
+class ChainRequestMatcher implements RequestMatcherInterface
+{
+    private $matchers;
+
+    /**
+     * @param RequestMatcherInterface[] $matchers
+     */
+    public function __construct(array $matchers)
+    {
+        $this->matchers = $matchers;
```

Comment:
You should loop through the matchers and maybe use a public `add` or `addMatcher()` which throws an invalid argument exception when it does not implement the `RequestMatcherInterface`.